src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -335,39 +329,13 @@ private async Task FormatDocumentCreatedFromTemplateAsync(IVsHierarchy hierarchy
             var forkedSolution = solution.AddDocument(DocumentInfo.Create(documentId, filePath, loader: new FileTextLoader(filePath, defaultEncoding: null), filePath: filePath));
             var addedDocument = forkedSolution.GetDocument(documentId)!;
 
+            // Call out to various new document formatters to tweak what they want
+            var formattingService = addedDocument.GetRequiredLanguageService<INewDocumentFormattingService>();
```

Comment:
what ensures that this exists?  for example, for F#/TS?

Reply:
Am I wrong in ascribing meaning to the absence of F#/TS implementations of `AbstractEditorFactory`? I guess they could use IVT.. It didn't seem like much of the other code was defensive though. Surely `document.GetRequiredSyntaxRoot` would fail for TS documents?

I can just make it resilient anyway though, this is the only spot that would need to change anyway.